### PR TITLE
USF-1657: Metadata and SEO

### PR DESCRIPTION
**Pull request by jeff-matthews**
This pull request (PR) adds implementation guidance to the base line **Metadata and SEO** docs in https://github.com/commerce-docs/microsite-commerce-storefront/pull/73.

Acceptance criteria:

- [x] document the script for bulk meta-data creation 
- [ ] ~outline how to correctly expose configurable products data for SEO purposes~ (will do this in a separate PR after details become clear)

I need help with configurable products. The internal wiki is not very clear.

### CODE REVIEW

**Review by herzog31 — APPROVED** [review]

File: src/content/docs/implementation/marketing/metadata.mdx [hunk 1]
```diff
@@ -1,56 +1,122 @@
...
+When importing pages, ensure that when a metadata block is present on a page, it does not contain any empty fields. This might lead to pages with missing titles, which Google penalizes.
+:::
+
+### Schema.org data (JSON-LD)
+
+JavaScript Object Notation for Linked Data (JSON-LD) is a structured data format that helps search engines understand the content of your web pages more explicitly. It is typically added as a script tag in the `<head>` of your HTML document.
+
+Adobe Commerce storefronts on Edge Delivery Services should include schema.org annotations to expose product data to search engines. This data should be included on all Commerce pages, especially PDPs and PLPs.
+
+You should compare the data available on the site before migrating to Edge Delivery Services to ensure that SKU's for all product variants are included. Use the [Rich Results Test](https://search.google.com/test/rich-results) to validate the schema.org annotations. The PDP dropin in the boilerplate contains an example for JSON-LD data.
```
Comment (keharper) [inline src/content/docs/implementation/marketing/metadata.mdx]:
```suggestion
You should compare the data available on the site before migrating to Edge Delivery Services to ensure that SKUs for all product variants are included. Use the [Rich Results Test](https://search.google.com/test/rich-results) to validate the schema.org annotations. The PDP dropin in the boilerplate contains an example for JSON-LD data.
```

File: src/content/docs/implementation/marketing/metadata.mdx [hunk 2]
```diff
@@ -1,56 +1,122 @@
...
+
+:::note
+When importing pages, ensure that when a metadata block is present on a page, it does not contain any empty fields. This might lead to pages with missing titles, which Google penalizes.
+:::
+
+### Schema.org data (JSON-LD)
+
+JavaScript Object Notation for Linked Data (JSON-LD) is a structured data format that helps search engines understand the content of your web pages more explicitly. It is typically added as a script tag in the `<head>` of your HTML document.
+
+Adobe Commerce storefronts on Edge Delivery Services should include schema.org annotations to expose product data to search engines. This data should be included on all Commerce pages, especially PDPs and PLPs.
```
Comment (keharper) [inline src/content/docs/implementation/marketing/metadata.mdx]:
```suggestion
Adobe Commerce storefronts on Edge Delivery Services should include [schema.org](https://schema.org/) annotations to expose product data to search engines. This data should be included on all Commerce pages, especially PDPs and PLPs.
```

File: src/content/docs/implementation/marketing/metadata.mdx [hunk 3]
```diff
@@ -1,56 +1,122 @@
...
+title: Indexing and metadata
+description: Boost your Adobe Commerce SEO on Edge Delivery Services! Follow our guidelines on data layers, canonical URLs, redirects, sitemaps, and metadata.
 ---
 
-Although Google is fine with metadata set by client-side JavaScript, social media platforms ignore it. For these platforms, metadata must be rendered server-side.
+import Aside from '@components/Aside.astro';
+import { Tabs, TabItem, Steps } from '@astrojs/starlight/components';
 
-Adobe recommends importing product metadata into Edge Delivery Services so that it can be rendered server-side on product detail pages. This is important so that Google Merchant Center can reliably verify entries from your product sheet. Also, social media sites, which don't usually parse JavaScript, can leverage this metadata to display rich previews of your product page links.
+This page provides guidelines for optimizing Adobe Commerce storefronts on Edge Delivery Services for SEO. It covers general recommendations for indexing and metadata.
```
Comment (keharper) [inline src/content/docs/implementation/marketing/metadata.mdx]: I don't like leading off with sentences like "This page/This section...", etc. If you could find a way to rewrite the above section, or the one for Indexing, that'd be great.

File: src/content/docs/implementation/marketing/metadata.mdx [hunk 4]
```diff
@@ -1,56 +1,122 @@
...
+### Multi-store setups
+
+For multi-store setups or stores supporting multiple locales:
+
+* Verify a [correct hreflang setup](https://www.aem.live/developer/sitemap#specifying-the-primary-language-manually). You can validate using the [hreflang Tags Testing Tool](https://technicalseo.com/tools/hreflang/)
+
+* Verify that the store code is part of the URL for every Catalog Service or Live Search request. This is required because Google caches Catalog Service responses without considering headers. Not adding the store code to the URL might lead to Google indexing the wrong data.
+
+  :::tip
+  You can ensure this by proxying the Catalog Service endpoint through Fastly. Alternatively, you can add a query parameter, but this method is incompatible with dropins.
```
Comment (keharper) [inline src/content/docs/implementation/marketing/metadata.mdx]: Add link to the Fastly doc here?

File: src/content/docs/implementation/marketing/metadata.mdx [hunk 5]
```diff
@@ -1,56 +1,122 @@
...
+  :::
+
+### Redirects
+
+Ensure proper redirects are set up using the redirects sheet or through your [CDN](/implementation/setup/content-delivery-network/).
+
+* Fallbacks for any potential URL changes (e.g. product URLs or missing .html suffix)
```
Comment (keharper) [inline src/content/docs/implementation/marketing/metadata.mdx]:
```suggestion
* Fallbacks for any potential URL changes (for example, product URLs or missing .html suffixes)
```

File: src/content/docs/implementation/marketing/metadata.mdx [hunk 6]
```diff
@@ -1,56 +1,122 @@
...
+
+You should compare the data available on the site before migrating to Edge Delivery Services to ensure that SKU's for all product variants are included. Use the [Rich Results Test](https://search.google.com/test/rich-results) to validate the schema.org annotations. The PDP dropin in the boilerplate contains an example for JSON-LD data.
+
+:::note
+If you are using Google Shopping features (for example, products available on `shopping.google.com`), consider schema.org annotations with critical priority. If a single JSON-LD annotation on the page is invalid, Google considers all annotations invalid.
+:::
+
+### Generate metadata
+
+You can use the [PDP Metadata Generator](https://github.com/hlxsites/aem-boilerplate-commerce/tree/main/tools/pdp-metadata) tool to automate the process of generating all recommended metadata. The tool is designed to fetch product data from the Catalog Service, process it, and generate a metadata spreadsheet in `.xslx` format, which can be used for the [bulk metadata](https://www.aem.live/docs/bulk-metadata) feature in Edge Delivery Services.
```
Comment (keharper) [inline src/content/docs/implementation/marketing/metadata.mdx]:
```suggestion
You can use the [PDP Metadata Generator](https://github.com/hlxsites/aem-boilerplate-commerce/tree/main/tools/pdp-metadata) tool to automate the process of generating all recommended metadata. The tool fetches product data from the Catalog Service, processes it, and generates a metadata spreadsheet in `.xslx` format. The spreadsheet can be used for the [bulk metadata](https://www.aem.live/docs/bulk-metadata) feature in Edge Delivery Services.
```

File: src/content/docs/implementation/marketing/metadata.mdx [hunk 7]
```diff
@@ -1,56 +1,122 @@
...
 
-### Installation
+To generate metadata using the PDP Metadata Generator tool:
 
-1. Clone the repository.
-1. Navigate to the project directory.
-1. Install the dependencies.
+<Steps>
+
+1. Clone the boilerplate repository (if you haven't already).
```
Comment (keharper) [inline src/content/docs/implementation/marketing/metadata.mdx]: This should be a prereq

File: src/content/docs/implementation/marketing/metadata.mdx [hunk 8]
```diff
@@ -1,56 +1,122 @@
...
    npm install
    ```
 
-### Configuration
+   :::note
+   Before running the tool, you must ensure that the `configFile` variable in the `pdp-metadata.js` file points to the correct configuration JSON file URL. This file contains the required parameters to access the Catalog Service API and should have been setup as part of your [project onboarding](/implementation/setup/commerce-configuration/).
+   :::
 
-Before running the tool, you must ensure that the `configFile` variable in the `pdp-metadata.js` file points to the correct configuration JSON file URL. This file contains the required parameters to access the Catalog Service and should have been setup as part of your project onboarding.
+1. Fetch product data, process it, and generate a file named `metadata.xlsx` in the project directory.
```
Comment (keharper) [inline src/content/docs/implementation/marketing/metadata.mdx]:
```suggestion
1. Run the tool and generate a file named `metadata.xlsx` in the project directory.
```

**Review by keharper — APPROVED** [review]

**Pull request merged**